src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsRecoveryObject.java

```diff
@@ -14,4 +21,41 @@ public String getRecoveredAdUnitId(String adUnitId) {
     jsActions.waitForJavaScriptTruthy(String.format("window._sp_ && window._sp_.getElementId && %s", getIdScript));
     return jsActions.execute(getIdScript).toString();
   }
+
+  public void verifyPageFairRecovery() {
+    By spansBodyChildrenSelector = By.cssSelector("body>span");
+    Dimension topLeaderboardSize = new Dimension(728, 90);
+    Dimension medrecSize = new Dimension(300, 250);
+
+    // when PF recovered ad is on page, inserts span elements as a direct children of body
+    wait.forElementPresent(spansBodyChildrenSelector);
+
+    // verify that adblock is turned on on that page
+    verifyNoAdsOnPage();
+
+    String firstSpanClass = driver.findElement(spansBodyChildrenSelector).getAttribute("class");
+    //System.out.println("\n\nCLASS:" + firstSpanClass);
+    List<WebElement> recoveredAds = driver
+        .findElements(By.cssSelector("body>span." + firstSpanClass))
+        .stream()
+        .filter(WebElement::isDisplayed)
+        .filter(e -> e.getCssValue("background").contains("data:image/jpeg"))
+        .collect(Collectors.toList());
+
+    String recoveredTopLeaderboard = "data:image/jpeg;base64,/9j/4QAYRXhpZgAASUkqAAgAAAAAAAAAAAAAAP/sABFEdWNreQABAAQAAABLAAD/4QPZaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wLwA8P3hwYWNrZXQgYmVnaW49Iu+7vyIgaWQ9Ilc1TTBNcENlaGlIenJlU3pOVGN6a2M5ZCI/PiA8eDp4bXBtZXRhIHhtbG5zOng9ImFkb2JlOm5zOm1ldGEvIiB4OnhtcHRrPSJBZG9iZSBYTVAgQ29";
+    String recoveredMedrec = "data:image/jpeg;base64,/9j/4QAYRXhpZgAASUkqAAgAAAAAAAAAAAAAAP/sABFEdWNreQABAAQAAAA3AAD/4QPZaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wLwA8P3hwYWNrZXQgYmVnaW49Iu+7vyIgaWQ9Ilc1TTBNcENlaGlIenJlU3pOVGN6a2M5ZCI/PiA8eDp4bXBtZXRhIHhtbG5zOng9ImFkb2JlOm5zOm1ldGEvIiB4OnhtcHRrPSJBZG9iZSBYTVAgQ29";
+
+    for (WebElement ad : recoveredAds) {
+      System.out.println("\n\nNA STRONIE:");
+      System.out.println(ad.getAttribute("class"));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS106)